libs/blocks/merch/merch.js

```diff
@@ -116,17 +124,62 @@ export async function getPriceContext(el, params) {
   };
 }
 
+let loadEntitlementsPromise;
+
+export async function loadEntitlements() {
+  await loadIms();
+  if (!window.adobeIMS?.isSignedInUser?.()) return Promise.resolve([{}, {}]);
+  const [{ default: getUserEntitlements },
+    mappings] = await Promise.all([
+    import('../global-navigation/utilities/getUserEntitlements.js'),
+    fetch(entitlementsURL),
+  ]);
+  const entitlements = await getUserEntitlements();
+  const entitlementsMetadata = await mappings.json();
+  return [entitlements, entitlementsMetadata];
+}
+
+export async function handleEntitlements(cta, promise) {
```

Comment:
Personal preference, but I usually steer away from using reserved keywords with different capitalisation, like `let Let`, this could cause confusion. Instead of just `promise`, maybe you could find a more specific name so it's immediately obvious what it represents

Reply:
I find `.promise`  very self describing but open to suggestions.